Assets/Scripts/TorchStand.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TorchStand : MonoBehaviour
+{
+    [SerializeField] private GameObject torchObject;
+    [SerializeField] private Light Light;
+    [SerializeField] private bool hasTorch = true;
+    [SerializeField] private bool isLightOn = false;
```

Comment:
This variables are kind need't and adding unnesecery complexity to code.
```cs
public bool IsTorchOn => light.enabled;
public bool HasTorch => torchObject.activeSelf;

public void TakeTorch()
    {
        torchObject.SetActive(false);
    }

public void PutTorch(bool isTorchLighted)
    {
        torchObject.SetActive(true);
        light.enabled = true;
    }
```
And now u can complity remove UpdateTorchStand and Update methods
Similar changesc can be made in Troch script.

If you want to call method from inspektor you can make
```cs
[ContextMenu("Method name in inspektor")]
private void A()
{
    Debug.Log("a");
}
```
and you will see it on the bottom
![image](https://github.com/user-attachments/assets/d182b3c0-bf92-45c3-8649-d18dbac3d54e)